doc/tutorials/run_all_tests_with_docker.md

```diff
@@ -0,0 +1,109 @@
+
+If you want to view all the available images, execute this command:
+```sh
+docker run --rm anoxis/registry-cli -r https://hub-public.libelektra.org
+```
+You will see something like this:
+
+```sh
+---------------------------------
+Image: build-elektra-debian-stretch
+  tag: 201906-ecf9161f41a8b472b3b0282a85a9f91d1f0f45357756e5451ae043fce8d0100e
+  tag: 201902-b6d49f470e1171348248b2f87ef397d58d7a2dae14d201f4073564079ce0c070
+  tag: 201903-b95dc56352aa684e16dfb8628bded4c69c712223f5d7ed99ebdd644852a32123
+  tag: 201905-ecf9161f41a8b472b3b0282a85a9f91d1f0f45357756e5451ae043fce8d0100e
+  tag: 201901-6b08855f13ba26e3ad1fa80e399b87df860cc24889f2d1854fa0050834567b26
+  tag: 201904-ecf9161f41a8b472b3b0282a85a9f91d1f0f45357756e5451ae043fce8d0100e
+  tag: 201904-1a6be7b9c3740a2338b14d08c757332cae5254ce58219b6cc2908c7bd6e4f460
+  tag: 201903-1a6be7b9c3740a2338b14d08c757332cae5254ce58219b6cc2908c7bd6e4f460
+  tag: 201903-6b08855f13ba26e3ad1fa80e399b87df860cc24889f2d1854fa0050834567b26
+  tag: 201902-6b08855f13ba26e3ad1fa80e399b87df860cc24889f2d1854fa0050834567b26
+..............................................................................
+```
+Afterwards pull your desired image as you would do from any public registry:
+```sh
+docker pull <image_name>:<tag_name>
+```
+
+EXAMPLE:
+```sh
+docker pull build-elektra-debian-stretch:201905-9dfe329fec01a6e40972ec4cc71874210f69933ab5f9e750a1c586fa011768ab
+```
+
+### 2. Run Docker contrainer
+
+You have to be in the root of the Elektra project, so that the container can properly map all the source files.
+
+So from your root project folder run the following:
+
+```sh
+docker run -it --rm \
+-v "$PWD:/home/jenkins/workspace" \
+-w /home/jenkins/workspace \
+<image_name>:<tag_name>
+```
+
+EXAMPLE:
+```sh
+docker run -it --rm \
+-v "$PWD:/home/jenkins/workspace" \
+-w /home/jenkins/workspace \
+ build-elektra-debian-stretch:201905-9dfe329fec01a6e40972ec4cc71874210f69933ab5f9e750a1c586fa011768ab
+```
+
+### 3. Build
+
+
+After starting the contrainer, you should be automatically inside it in the working directory we set --> /home/jenkins/workspace
+
+Create folder for building project
+
+```sh
+mkdir build-docker && cd build-docker
+```
+
+Build it with
+
+```sh
+ cmake .. \
+-DBINDINGS="ALL;-DEPRECATED;-haskell" \
+-DPLUGINS="ALL;-DEPRECATED" \
+-DTOOLS="ALL" \
+-DENABLE_DEBUG=ON \
+-DKDB_DB_HOME="$PWD" \
+-DKDB_DB_SYSTEM="$PWD/.config/kdb/system" \
+-DKDB_DB_SPEC="$PWD/.config/kdb/system"
+```
+and then with
+```sh
+make -j 10
+```
+The number 10 can be changed as follows: number of supported simultaneous threads by your CPU + 2. But don't worry, this can only affect the speed of the building, it cannot really break it.
+### 4. Run tests
+
+Finally run the tests
+```sh
+make run_all
+```
```

Comment:
Also describe how to install Elektra and run the installed tests (`kdb run_all`). Please also link to doc/TESTING.md.

Reply:
What do you mean? Elektra is built within the container anyway. Why do we need to install it?

Reply:
To also install Elektra `make install` and run then the installed tests. The build server also does this and it is possible that the installed tests have different results (e.g. if test data is missing). See scripts/jenkins/Jenkinsfile

Reply:
Oh, I didn't know it. So basically before executing `make run_all` `make install` should be executed first?

Reply:
Yes, some tests only run if Elektra is installed.

And there is a second set of tests to be executed with:

```sh
kdb run_all
```

These tests also need an installed Elektra.

If your tutorial should cover how to execute *all* tests, you should mention this.